### PR TITLE
[DEV-6449] Fix sorting error

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Fixes a 500 error when sorting by `recent_publication_date` on the `/v2/reporting/agencies/overview/` endpoint.

**Technical details:**
Moves the sorting from python to postgres. 

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend
4. [N/A] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-6449](https://federal-spending-transparency.atlassian.net/browse/DEV-6449):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/reporting/v2/views/agencies/overview.py [hunk 1]
```diff
@@ -106,17 +108,24 @@ def get_agency_overview(self):
...
                 "tas_obligations",
                 "tas_obligation_not_in_gtas_total",
-                "missing_tas_accounts",
+                "missing_tas_accounts_count",
+                "fiscal_year",
+                "fiscal_period",
+                "missing_tas_accounts_count",
                 "fiscal_year",
                 "fiscal_period",
                 "submission_is_quarter",
```
Comment (tony-sappe) [inline usaspending_api/reporting/v2/views/agencies/overview.py:123]: Several duplicate values in this list

File: usaspending_api/reporting/v2/views/agencies/overview.py [hunk 2]
```diff
@@ -106,17 +108,24 @@ def get_agency_overview(self):
                 "abbreviation",
                 "toptier_code",
```
Comment (tony-sappe) [inline usaspending_api/reporting/v2/views/agencies/overview.py:109]: Sort value `agency_code` is listed in the API contract and sortable fields but is a 500 error
Sort value `missing_tas_accounts_total` is listed in the contract but is not accepted by the API (now `missing_tas_accounts_count` in the code)

**Review by tony-sappe — CHANGES_REQUESTED** [review]
Inconsistencies with sort values and duplicates in queryset

File: usaspending_api/reporting/v2/views/agencies/overview.py (hunk 2 above)
Comment (emily-brents) [reply usaspending_api/reporting/v2/views/agencies/overview.py:112]: Yeah, there was some confusion originally over what was supposed to be there, but I'm just gonna include both.

File: usaspending_api/reporting/v2/views/agencies/overview.py (hunk 1 above)
Comment (emily-brents) [reply usaspending_api/reporting/v2/views/agencies/overview.py]: That's what happens when you don't look at merge conflicts that closely

File: usaspending_api/reporting/v2/views/agencies/overview.py [hunk 3]
```diff
@@ -106,20 +108,18 @@ def get_agency_overview(self):
             .values(
                 "agency_name",
                 "abbreviation",
+                "agency_code",
                 "toptier_code",
```
Comment (tony-sappe) [inline usaspending_api/reporting/v2/views/agencies/overview.py:112]: Is `toptier_code` still needed if `agency_code` is being used?

Comment (emily-brents) [reply usaspending_api/reporting/v2/views/agencies/overview.py:112]: `toptier_code` is used by `create_assurance_statement_url`, which is also used by some other endpoints that use `toptier_code`. It seemed better to include `toptier_code` than to re-write that funtion

Comment (tony-sappe) [reply usaspending_api/reporting/v2/views/agencies/overview.py:112]: OK, thanks for explaining

**Review by tony-sappe — APPROVED** [review]

**Pull request merged**